### DISCUSSION TITLE
Get cpEditUrl through GraphQL or construct it with props

### DISCUSSION THREAD

**Discussion by sgtpenguin** [craftcms/cms#9983] [Ideas] [answered]
I've been tasked with adding an "Edit" link on the frontend of a headless build for logged in users.  I can determine if someone is logged in with:
`/index.php?p=admin/actions/users/session-info&dontExtendSession=1`

I can't figure out how to either get the full `cpEditUrl` through graphql, or how to make it using the element id or some other prop.

**Comment by brandonkelly** [comment, marked as answer]
Really all you need is the entry’s ID, and then you can construct a URL to `/admin/edit/EntryID`. Craft will redirect the browser to the actual entry edit URL from there.

**Reply by sgtpenguin** [reply]
perfect.  Thanks @brandonkelly
